build.sh

```diff
@@ -111,45 +139,79 @@ inject_metadata () {
 # Main entry point.
 # ------------------------------------------------
 
-# Make sure we start from a clean slate.
-rm -rf built
-rm -rf downloads_cache
-rm -f master.tar.gz
-rm -f metadata.tar.gz
-
-# Run basic tests.
-echo "Running basic sanity tests on metadata."
-echo "If these fail, then fix whatever is causing them first."
+if [ -n "$ghprbActualCommit" ]
+then
+    echo "Commit hash: $ghprbActualCommit"
+    export COMMIT_CHANGES="`git diff --diff-filter=AM --name-only --stat origin/master`"
+else
+    echo "No commit provided, skipping further tests."
+    exit $EXIT_OK
+fi
 
-if ! prove
+# Make sure we start from a clean slate.
+if [ -d "built/" ]
 then
-    echo "Prove step failed."
-    exit 1
+    rm -rf built
 fi
 
-# Find the changes to test.
-echo "Finding changes to test..."
+if [ -d "downloads_cache/" ]
+then
+    rm -rf downloads_cache
+fi
 
-if [ -z $ghprbActualCommit ]
+if [ -e "master.tar.gz" ]
 then
-    echo "No commit hash, running all netkan files"
-    export COMMIT_CHANGES=NetKAN/*.netkan
-else
-    echo "Commit hash: $ghprbActualCommit"
-    export COMMIT_CHANGES="`git diff --diff-filter=AM --name-only --stat origin/master NetKAN`"
+    rm -f master.tar.gz
 fi
 
-if [ "$COMMIT_CHANGES" = "" ]
+if [ -e "metadata.tar.gz" ]
 then
-    echo "No .netkan changes, skipping further tests."
-    exit 0
+    rm -f metadata.tar.gz
 fi
 
```

Comment:
These are mainly for local testing, we'll be getting a clean container on every build. The checks mean we'll only try to remove things when they exist.